### PR TITLE
Postgres - killbill-commons - 1 of 4

### PR DESCRIPTION
**Pull request by marksimu**
@dconcha @mhmartinez 

Support for targeting Postgres.

### CODE REVIEW

File: jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java [hunk 1]
```diff
@@ -48,10 +48,9 @@ public LocalDateArgument(final LocalDate value) {
         @Override
         public void apply(final int position, final PreparedStatement statement, final StatementContext ctx) throws SQLException {
             if (value != null) {
-                // ISO8601 format
-                statement.setString(position, value.toString());
+                statement.setDate(position, new java.sql.Date(value.toDate().getTime()));
```
Comment (dconcha) [inline jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java:51]: are there any potential side effects of this change for other engines such as Mysql?

Comment (marksimu) [inline jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java:51]: I did not observe any, under MySQL and H2.

Comment (dconcha) [inline jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java:51]: I think they should be apparent when running tests, but i had to ask :)

**Comment by dconcha** [comment]
LGTM

File: jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java (hunk 1 above)
Comment (sbrossie) [inline jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java:51]: That makes me extremely nervous! We had a bug with the way we stored date in the DB which if remember correctly lead us to use that approach. I spent an entire week chasing the bug... why do we need the change?

File: jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java [hunk 2]
```diff
@@ -48,10 +48,9 @@ public LocalDateArgument(final LocalDate value) {
...
             } else {
-                statement.setNull(position, Types.VARCHAR);
+                statement.setNull(position, Types.DATE);
             }
```
Comment (sbrossie) [inline jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java:54]: Why that change?

File: jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java (hunk 1 above)
Comment (marksimu) [inline jdbi/src/main/java/org/killbill/commons/jdbi/argument/LocalDateArgumentFactory.java:51]: String parameters do not implicitly cast: "org.postgresql.util.PSQLException: ERROR: column "invoice_date" is of type date but expression is of type character varying"

I started with a Postgres-specific solution (https://github.com/marksimu/killbill-commons/commit/cfed19d889a21191928dc0fc47e44e8d758b6991) but that seemed to be hacky and unnecessary.  Will look into this further.

**Pull request merged**